xbmc/games/addons/GameClientProperties.cpp

```diff
@@ -289,3 +320,111 @@ bool CGameClientProperties::HasProxyDll(const std::string& strLibPath) const
   }
   return false;
 }
+
+bool CGameClientProperties::InstallDependencies(const std::vector<std::string>& addons)
+{
+  // Only proceed if we have a GUI
+  CGUIComponent* const gui = CServiceBroker::GetGUI();
+  if (gui == nullptr)
+    return false;
+
+  const CGUIWindowManager& windowManager = gui->GetWindowManager();
+
+  // Get GUI dialogs
+  auto* selectDialog = windowManager.GetWindow<CGUIDialogSelect>(WINDOW_DIALOG_SELECT);
+  auto* progressDialog = windowManager.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS);
+
+  // We can only install add-ons if the dialogs are present
+  if (selectDialog == nullptr || progressDialog == nullptr)
+    return false;
+
+  // Get installable add-ons
+  VECADDONS installableAddons;
+  for (const std::string& dependency : addons)
+  {
+    AddonPtr addon;
+    if (CServiceBroker::GetAddonMgr().FindInstallableById(dependency, addon) && addon)
+      installableAddons.emplace_back(std::move(addon));
+    else
+      CLog::Log(LOGERROR, "Failed to find installable add-on for {}", dependency);
+  }
+
+  // Verify all add-ons are installable
+  if (addons.size() != installableAddons.size())
+    return false;
+
+  // Setup select dialog
+  selectDialog->Reset();
+  selectDialog->SetHeading(39020); // "The following additional add-ons will be installed"
+  selectDialog->SetUseDetails(true);
+  selectDialog->EnableButton(true, 186); // "OK""
+  selectDialog->SetButtonFocus(true);
+  for (const auto& addon : installableAddons)
+  {
+    CFileItem item{addon->Name()};
+    item.SetArt("icon", addon->Icon());
+    selectDialog->Add(item);
```

Comment:
Could `std::move(item)` be senseful here?

Reply:
I originally used std::move here, but the dialog only accepts a const ref, so we'd either have to modify the dialog or drop the move operator. I chose the second.

Reply:
Alright. Let’s keep it as is.